internal/component/otelcol/connector/count/count.go

```diff
@@ -0,0 +1,195 @@
+}
+
+// Convert implements connector.Arguments.
+func (args Arguments) Convert() (otelcomponent.Config, error) {
+	var (
+		spans      = convertMetricInfo(args.Spans)
+		spanEvents = convertMetricInfo(args.SpanEvents)
+		metrics    = convertMetricInfo(args.Metrics)
+		dataPoints = convertMetricInfo(args.DataPoints)
+		logs       = convertMetricInfo(args.Logs)
+	)
+	cfg := &countconnector.Config{
+		Spans:      spans,
+		SpanEvents: spanEvents,
+		Metrics:    metrics,
+		DataPoints: dataPoints,
+		Logs:       logs,
+	}
+	if err := checkConverted(args, cfg); err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+func checkConverted(args Arguments, cfg *countconnector.Config) error {
+	var err error
+	err = multierr.Append(err, dupeCheck("span", args.Spans, cfg.Spans))
+	err = multierr.Append(err, dupeCheck("spanevent", args.SpanEvents, cfg.SpanEvents))
+	err = multierr.Append(err, dupeCheck("metric", args.Metrics, cfg.Metrics))
+	err = multierr.Append(err, dupeCheck("datapoint", args.DataPoints, cfg.DataPoints))
+	err = multierr.Append(err, dupeCheck("log", args.Logs, cfg.Logs))
+	return err
+}
+
+func dupeCheck(signalType string, mi []MetricInfo, ccmi map[string]countconnector.MetricInfo) error {
+	var err error
+	if len(mi) != len(ccmi) {
+		nameMap := make(map[string]struct{})
+		for _, entry := range mi {
+			if _, ok := nameMap[entry.Name]; ok {
+				err = multierr.Append(err, fmt.Errorf("duplicate %s name: %s", signalType, entry.Name))
+			} else {
+				nameMap[entry.Name] = struct{}{}
+			}
+		}
+	}
+	return err
+}
+
+func convertMetricInfo(mi []MetricInfo) map[string]countconnector.MetricInfo {
+	ret := make(map[string]countconnector.MetricInfo)
+	for _, metricInfo := range mi {
+		var attrConfigs []countconnector.AttributeConfig
+		for _, ac := range metricInfo.Attributes {
+			a := countconnector.AttributeConfig{
+				Key:          ac.Key,
+				DefaultValue: ac.DefaultValue,
+			}
+			attrConfigs = append(attrConfigs, a)
+		}
+		ret[metricInfo.Name] = countconnector.MetricInfo{
+			Description: metricInfo.Description,
+			Conditions:  metricInfo.Conditions,
+			Attributes:  attrConfigs,
+		}
+	}
+	return ret
+}
+
+// Exporters implements connector.Arguments.
+func (Arguments) Exporters() map[otelcomponent.Type]map[otelcomponent.ID]otelcomponent.Component {
+	return nil
+}
+
+// Extensions implements connector.Arguments.
+func (Arguments) Extensions() map[otelcomponent.ID]otelcomponent.Component {
+	return nil
+}
+
+// NextConsumers implements connector.Arguments.
+func (args Arguments) NextConsumers() *otelcol.ConsumerArguments {
+	return args.Output
+}
+
+// Validate implements river.Validator.
+func (args *Arguments) Validate() error {
+	cfg, err := args.Convert()
+	if err != nil {
+		return err
+	}
+	return cfg.(*countconnector.Config).Validate()
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = DefaultArguments
```

Comment:
If there is more than one such connector in the config file, I wonder if there'd be a bug due to reusing the same `DefaultArguments`. E.g. if one connector overrides some defaults, but another one doesn't, then would the connector which only used defaults pick up the settings from the other one? I wonder if we should copy the slices with defaults every time we call `SetToDefault`

Reply:
I can test this out. I noticed several other components following this pattern though, I'd assume they have the same issue if this is the case.